tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/unit_normalization_test.py

```diff
@@ -0,0 +1,84 @@
+import os
+import unittest
+
+from google.api_core.exceptions import NotFound
+from jinja2 import Environment
+
+import tests.test_util as test_util
+from app_identity import PROJECT_ID
+from cdr_cleaner.cleaning_rules.unit_normalization import UnitNormalization, UNIT_MAPPING_TABLE
+# Project Imports
+from utils import bq
+
+jinja_env = Environment(
+    # help protect against cross-site scripting vulnerabilities
+    autoescape=True,
+    # block tags on their own lines
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
+
+test_query = jinja_env.from_string("""select * from `{{intermediary_table}}`""")
+
+
+class UnitNormalizationTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        # Set the test project identifier
+        project_id = os.environ.get(PROJECT_ID)
+        self.project_id = project_id
+
+        # Set the expected test datasets
+        self.dataset_id = os.environ.get('COMBINED_DATASET_ID')
+        self.sandbox_id = self.dataset_id + '_sandbox'
+
+        self.query_class = UnitNormalization(self.project_id, self.dataset_id,
+                                             self.sandbox_id)
+
+        self.assertEqual(self.query_class.get_project_id(), self.project_id)
+        self.assertEqual(self.query_class.get_dataset_id(), self.dataset_id)
+        self.assertEqual(self.query_class.get_sandbox_dataset_id(),
+                         self.sandbox_id)
+
+    def test_setup_rule(self):
+
+        # test if intermediary table exists before running the cleaning rule
+        intermediary_table = f'{self.project_id}.{self.dataset_id}.{UNIT_MAPPING_TABLE}'
+
+        client = bq.get_client(self.project_id)
+
+        def table_exists(bq_client, table):
+            try:
+                bq_client.get_table(table)
+            except NotFound:
+                return
+
+        self.assertRaises(NotFound, table_exists(client, intermediary_table))
```

Comment:
I contest the value of this part of the test- whether or not the lookup exists in the test environment is up to how we have set it up. We should be sure it doesn't exist.

Reply:
I agree.